openmc/tallies.py

```diff
@@ -8,6 +8,7 @@
 import sys
 
 import numpy as np
+import h5py
```

Comment:
Did we want to keep `import h5py` next to its use? I know someone opined at some point that having the import at the module-level makes the entire API depend on it, so if you don't have h5py you can't even construct inputs.